2024/YoctoProjectDeveloperDay/practical-sec-tee.md

```diff
@@ -0,0 +1,994 @@
+    - Present linkinng binaries vs ELF using Binman.
+    - Present `binman`-related problems.
+
+---
+
+# Linux integration
+
+Turn on `CONFIG_TEE` and `CONFIG_OPTEE` (and related, if needed) configs, and
+add OP-TEE OS and its memory to DTS (here are `CFG_TZDRAM_START`,
+`CFG_TZDRAM_SIZE`, `CFG_SHMEM_START` and `CFG_SHMEM_SIZE`):
+
+```dts
+    reserved-memory {
+		#address-cells = <2>;
+		#size-cells = <2>;
+		ranges;
+
+		optee@8400000 {
+			reg = <0x0 0x08400000 0x0 0x02000000>;
+            no-map;
+		};
+
+		optee_shm: optee_shared_mem@a400000 {
+			reg = <0x0 0x0a400000 0x0 0x00400000>;
+		};
+	};
+
+	firmware {
+		optee: optee {
+			compatible = "linaro,optee-tz";
+			method = "smc";
+			shm = <&optee_shm>;
+		};
+    };
+```
+
+???
+
+- Time for this slide: TODO
+- Idea/goal of this slide: A list of steps on how to integrate ATF and OP-TEE
+  into Linux.
+- What to say:
+    - TODO
+- Notes:
+    - Present some Linux Kernel configs;
+    - Present devicetree modifications;
+    - Some userspace tools and libs integration.
+
+---
+
+# Adding packages and recipes:
+
+1. Add TFA `.bbappend` with configuration according to
+  `meta-arm/recipes-bsp/trusted-firmware-a/trusted-firmware-a_2.10.3.bb`;
+2. Add OP-TEE OS `.bbappend` with configuration according to
+  `meta-arm-bsp/recipes-bsp/trusted-firmware-a/trusted-firmware-a_%.bbappend`;
+3. You can add OP-TEE OS testsuite according to
+  `meta-arm-bsp/recipes-security/optee/optee-test_4.%.bbappend`;
+4. Add the packages to your packagegroup:
+
+    ```bb
+    RDEPENDS:${PN}-security = " \
+        trusted-firmware-a \
+        optee-client \
+        optee-os \
+        optee-test \
+        libp11 \
+    "
+    ```
+???
+
+- Time for this slide: TODO
+- Idea/goal of this slide: TODO
+- What to say:
+    - TODO
+- Notes:
+    - TODO
+
+---
+
+# OP-TEE Secure Storage
+
+.center[ <img src="./img/op-tee-secure-storage.svg" height="350px"> ]
+<br>
+.center[ Configured during OP-TEE OS compilation via `CFG_RPMB_FS` and `CFG_REE_FS`. ]
+
+???
+
+- Time for this slide: TODO
+- Idea/goal of this slide: TODO
+- What to say:
+    - TODO
+- Notes:
+    - TODO
+
+---
+
+# Testing using vendor binaries
+
+```bash
+(...)
+U-Boot SPL 2024.01 (Jan 08 2024 - 15:37:48 +0000)
+(...)
+NOTICE:  BL31: v2.3():v2.3-607-gbf602aff1:cl
+NOTICE:  BL31: Built : 10:16:03, Jun  5 2023
+(...)
+I/TC: OP-TEE version: 3.13.0-723-gdcfdd61d0 #hisping.lin (gcc version 10.2.1...
+(...)
+root@quartz64-a:~# dmesg
+(...)
+root@quartz64-a:~# alias p11="pkcs11-tool --module /usr/lib/libckteec.so.0"
+root@quartz64-a:~# p11 --show-info
+E/LD:  ta_elf_load_main:1128 sys_map_zi stack
+E/TC:? 0 ldelf_init_with_ldelf:126 ldelf failed with res: 0xffff000c
+ERR [291] LT:ckteec_invoke_init:304: TEEC open session failed ffff000c from 3
+(...)
+root@quartz64-a:~# xtest
+(...)
+653 subtests of which 188 failed
+106 test cases of which 76 failed
+0 test cases were skipped
+TEE test application done!
+```
```

Comment:
That's a lot of information for one slide and it's a little unreadable. Maybe split it into shorter slides and just say what, where and why you're running?

Reply:
:thinking:

Reply:
Fixed here 0de9c5b7e32ff2a61a413d9a8cd2bd4c2fab964e.